core/execution/future/market.go

```diff
@@ -3199,9 +3445,19 @@ func (m *Market) amendOrder(
 	if expiryChange || sizeDecrease || timeInForceChange || icebergSizeIncrease {
 		ret := m.orderAmendInPlace(existingOrder, amendedOrder)
 		if sizeDecrease {
-			// ensure we release excess if party reduced the size of their order
-			m.recheckMargin(ctx, m.position.GetPositionsByParty(amendedOrder.Party))
+			if m.getMarginMode(party) == types.MarginModeCrossMargin {
+				// ensure we release excess if party reduced the size of their order
+				m.recheckMargin(ctx, m.position.GetPositionsByParty(amendedOrder.Party))
+			}
+		}
+		if m.getMarginMode(party) == types.MarginModeIsolatedMargin {
+			pos, _ := m.position.GetPositionByPartyID(amendedOrder.Party)
+			if err := m.updateIsolatedMarginOnOrder(ctx, pos, amendedOrder); err == risk.ErrInsufficientFundsForMarginInGeneralAccount {
```

Comment:
Same thing here, that sounds weird to me, but this might be per spec, I haven't checked for this specifically.

Reply:
well then if you did check the spec you'd see that:

<img width="933" alt="image" src="https://github.com/vegaprotocol/vega/assets/6655556/dbffe4ea-53c1-46dc-b92f-08436db44242">